drivers/overlay/overlay.go

```diff
@@ -309,9 +309,10 @@ func Init(home string, options graphdriver.Options) (graphdriver.Driver, error)
 	if err != nil {
 		return nil, err
 	}
-	if fsName, ok := graphdriver.FsNames[fsMagic]; ok {
-		backingFs = fsName
+	if fsName, ok := graphdriver.FsNames[fsMagic]; !ok {
+		return nil, fmt.Errorf("filesystem type %#x reported for %s is not supported with 'overlay': %w", fsMagic, filepath.Dir(home), graphdriver.ErrIncompatibleFS)
 	}
+	backingFs = fsName
```

Comment:
Please move this to an `else` block, since `fsName` does not exists outside of the scope.

Reply:
I agree. That was a mistake.